os/mm/mm_heap/mm_initialize.c

```diff
@@ -146,13 +146,79 @@ void mm_addregion(FAR struct mm_heap_s *heap, FAR void *heapstart, size_t heapsi
 	heapinfo_update_node((FAR struct mm_allocnode_s *)heap->mm_heapstart[IDX], 0xDEAD);
 #endif
 
+#ifdef CONFIG_MM_REALTIME_SUPPORT
+	int i = 0;
+	int j;
+	int seg_size = MM_REALTIME_SUPPORT_MIN_SIZE; /* minimum size for realtime support */
+	int mem_pool_size = 0;
+	int prev_size;
+	char *numbers = CONFIG_MM_REALTIME_SUPPORT_NUMS;
+	char *ptr;
+
+	while (*numbers != '\0') {
+		heap->mm_realtime_num[i] = (int)strtol(numbers, &ptr, 10);
+		while (*ptr == ',' || *ptr == ' ') {
+			++ptr;
+		}
+		numbers = ptr;
+		if (++i >= MM_REALTIME_SUPPORT_NUMOF_SIZES) {
+			break;
+		}
+	}
+
+	for (i = 0; i < MM_REALTIME_SUPPORT_NUMOF_SIZES; ++i) {
+		if (heap->mm_realtime_num[i] < 1) {
+			heap->mm_realtime_num[i] = 0;
+		}
+		mem_pool_size += (heap->mm_realtime_num[i] * (i + 2) * MM_REALTIME_SUPPORT_STEP_SIZE);
+		llvdbg("%d free nodes for size %d are reserved.\n", heap->mm_realtime_num[i], (i + 2) * MM_REALTIME_SUPPORT_STEP_SIZE);
+
+	}
+	llvdbg("mm_pool_size = %d\n", mem_pool_size);
+
+	/* First, add one big chunk of free node for non-realtime. */
+	node            = (FAR struct mm_freenode_s *)(heapbase + SIZEOF_MM_ALLOCNODE);
+	node->size      = heapsize - 2 * SIZEOF_MM_ALLOCNODE - mem_pool_size;
```

Comment:
You had better add bracket for multiplication.

Reply:
I don't think it's necessary, but fixed.

Reply:
yes minor one, just for readability, thanks.